### DISCUSSION TITLE
integration functioning

### DISCUSSION THREAD

**Discussion by matly006** [djansen1987/SAJeSolar#68] [Q&A] [answered]
Hi, I apologize in advance for the English. I installed the new version and modified my configuration.yaml file as indicated, inserting the URL of my portal into the domain which corresponds to: esaj-home.saj-electric.com. however I cannot receive the data. From the application log I get this:

Logger: custom_components.saj_esolar.sensor
Source: custom_components/saj_esolar/sensor.py:877
Integration: saj_esolar (documentation, issues)
First occurred: 16:44:29 (475 occurrences)
Last logged: 16:56:29

return data NONE

and also this:

Logger: custom_components.saj_esolar.sensor
Source: custom_components/saj_esolar/sensor.py:876
Integration: saj_esolar (documentation, issues)
First occurred: 16:49:32 (2 occurrences)
Last logged: 16:54:32

Cannot poll eSolar using url: %s

do I need to change anything else or where am I wrong? Thanks in advance

**Comment by djansen1987** [comment]
Hi the URL you are referring to is the new portal url, that one is not yet supported as stated in my release notes. Which brand do you have?

**Reply by matly006** [reply]
both batteries and inverter are branded saj

**Reply by djansen1987** [reply]
And this stopped working since last month? Because this should still work without the extra configuration options

**Reply by matly006** [reply]
I honestly installed it last week. Previously I had integration via Modbus active but the assistance service decided to update the inverter software and therefore it no longer worked. I was hoping that with this I would succeed but........

However, if you think it should work anyway, I'll have some other problems. Do you have any suggestions on where to check?

Thanks again

**Reply by djansen1987** [reply]
Can you see your data in the old portal?

Also can you share you you config(without username and password)

**Reply by matly006** [reply, marked as answer]
Hi, yes I can see the data on the old portal.

However I solved it.
Unfortunately I am reluctant to make updates on things that work, however this time it was done. I updated home assistat, removed the integration, cleaned up my cofiguration.yaml and rebooted the system. then installed the integration again and now it works.

Thank you for your availability

**Reply by djansen1987** [reply]
Good to hear! Almost thought there was another issue with the new portal

**Comment by Txetxo1979** [comment]
Hello, I'm sorry, but I must be very clumsy. When I check the configuration to restart Homa Assistant with the new URL included in sensor.py I get the following error:
Platform error sensor.saj_esolar - Exception importing custom_components.saj_esolar.sensor

My URL is SolarProfit, https://inversor.saj-electric.com/cloud

I don't know what I'm doing wrong, since I also have the URL configured in the Configuration.yaml:
  - platform: saj_esolar
    username: abcde
    password: 1234567890
    provider_domain: investor.saj-electric.com
    provider_path: cloud
    provider_ssl: False
    resources:
      - nowPower
      - runningState
      - todayElectricity
      - monthElectricity
      - yearElectricity
      - totalElectricity # Energy -> Solar production
      - todayGridIncome
      - income
      - lastUploadTime
      - totalPlantTreeNum
      - totalReduceCo2
      - isAlarm # Yes / No
      - status
      - plantuid
      -currency
      - address
      - isOnline
      - peakPower
      - systemPower # Installed capacity


The bad thing is that I have batteries and I won't be able to see their status.

Can you help me figure out what I'm doing wrong?

Thank you

**Reply by djansen1987** [reply]
hi with the latest version you don't have to change anything in sensor.py. please revert the sensor.py and use above config (typo in -currency?)
